### PR TITLE
transformquery: fix usage of deprecated func

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: pkg/telemetryquerylanguage/contexts/metrics/metrics.go [hunk 1]
```diff
@@ -807,15 +807,15 @@ func accessPositiveBucketCounts() pathGetSetter {
...
+				return ctx.GetItem().(pmetric.ExponentialHistogramDataPoint).Positive().BucketCounts().AsRaw()
 			}
 			return nil
 		},
 		setter: func(ctx tql.TransformContext, val interface{}) {
 			if newPositiveBucketCounts, ok := val.([]uint64); ok {
 				switch ctx.GetItem().(type) {
 				case pmetric.ExponentialHistogramDataPoint:
-					ctx.GetItem().(pmetric.ExponentialHistogramDataPoint).Positive().SetMBucketCounts(newPositiveBucketCounts)
+					ctx.GetItem().(pmetric.ExponentialHistogramDataPoint).Positive().SetBucketCounts(pcommon.NewImmutableUInt64Slice(newPositiveBucketCounts))
```
Comment (bogdandrutu) [inline pkg/telemetryquerylanguage/contexts/metrics/metrics.go:818]: @TylerHelmuth these are unnecessary extra copies during the call to "NewImmutable", I know you did some smartness for TraceID, we should do the same for this, in a separate PR.

**Review by dmitryax — APPROVED** [review]

File: pkg/telemetryquerylanguage/contexts/metrics/metrics.go (hunk 1 above)
Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/contexts/metrics/metrics.go:818]: Good idea, can you open an issue with the details?

Comment (bogdandrutu) [reply pkg/telemetryquerylanguage/contexts/metrics/metrics.go:818]: I was trying to delegate that to you :))

**Review by TylerHelmuth — APPROVED** [review]

File: pkg/telemetryquerylanguage/contexts/metrics/metrics.go (hunk 1 above)
Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/contexts/metrics/metrics.go:818]: image

Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/contexts/metrics/metrics.go:818]: I'll take it since I own the package.

Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/contexts/metrics/metrics.go:818]: done: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12872

**Pull request merged**